src/Models/Booking.php

```diff
@@ -5,85 +5,31 @@
 namespace Tipoff\Bookings\Models;
 
 use Carbon\Carbon;
-use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Tipoff\Support\Models\BaseModel;
+use Tipoff\Support\Traits\HasCreator;
+use Tipoff\Support\Traits\HasPackageFactory;
+use Tipoff\Support\Traits\HasUpdater;
 
 class Booking extends BaseModel
 {
-    use HasFactory;
-
-    protected $casts = [
-    ];
+    use HasPackageFactory;
+    use HasCreator;
+    use HasUpdater;
 
     protected $with = [
+        'variation',
+        'experience',
         'order',
-        'slot',
+        'agent',
+        'user',
+        'subject',
     ];
 
-    protected $appends = [
-        'date',
-    ];
-
-    protected static function boot()
-    {
-        parent::boot();
-
-        static::saving(function ($booking) {
-            if (empty($booking->order_id)) {
-                throw new \Exception('A booking must be part of an order.');
-            }
-            if (empty($booking->slot_id)) {
-                throw new \Exception('A booking must be for an availability slot.');
-            }
-        });
-    }
-
-    /**
-     * Generate amount, total_taxes and total_fees.
-     *
-     * !!! This funcionaly is uswed mainly by seeder !!!
-     *
-     * @return void
-     */
-    public function generatePricing()
```

Comment:
I removed a significant portion of the existing methods. Not sure how many are still relevant based on the updated model structure.